src/Controller/HearingController.php

```diff
@@ -462,4 +495,48 @@ public function resumeHearing(CaseEntity $case, Hearing $hearing): Response
 
         return $this->redirectToRoute('case_hearing_index', ['id' => $case->getId()]);
     }
+
+    /**
+     * @Route("/{case}/hearing/{hearingPost}/delete", name="case_hearing_post_delete")
+     */
+    public function hearingPostDelete(CaseEntity $case, HearingPost $hearingPost, DocumentUploader $documentUploader, CaseDocumentRelationRepository $relationRepository, Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('edit', $case);
+
+        // Check that CSRF token is valid
+        if ($this->isCsrfTokenValid('delete'.$hearingPost->getId(), $request->request->get('_token'))) {
+            // Remove relation between case and document
+            $document = $hearingPost->getDocument();
+            $relation = $relationRepository->findOneBy(['case' => $case, 'document' => $document]);
+
+            if (null !== $relation) {
+                $this->entityManager->remove($relation);
+                $this->entityManager->flush();
+            }
+
+            // Remove file
+            $documentUploader->deleteDocumentFile($document);
+
+            // Set foreign keys between hearing post and document to null
+            $hearingPost->getDocument()->setHearingPost(null);
+            $hearingPost->setDocument(null);
+            $this->entityManager->flush();
+
+            // Remove new hearing post alert
+            $hearingPost->getHearing()->setHasNewHearingPost(false);
+
+            // Remove hearing post and document
+            $this->entityManager->remove($hearingPost);
+            $this->entityManager->remove($document);
+            $this->entityManager->flush();
+
+            if ($hearingPost instanceof HearingPostResponse) {
+                $this->addFlash('success', new TranslatableMessage('Hearing post response deleted', [], 'case'));
+            } elseif ($hearingPost instanceof HearingPostRequest) {
+                $this->addFlash('success', new TranslatableMessage('Hearing post request deleted', [], 'case'));
+            }
```

Comment:
See https://github.com/itk-dev/naevnssekretariatet/blob/develop/src/Controller/AgendaItemController.php#L68-L72 for another (better?) way to do this.

Reply:
Refactored to use proposed method.